### ISSUE TITLE
azurerm_kubernetes_cluster.dns_prefix should be required

### ISSUE THREAD

**Issue by NickMRamirez** [terraform-providers/terraform-provider-azurerm#1306]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.11.3

### Affected Resource(s)

* azurerm_kubernetes_cluster

### Terraform Configuration Files

Note that I have not set the `dns_prefix` field on the `azurerm_kubernetes_cluster` resource.

```hcl
resource "azurerm_kubernetes_cluster" "aks" {
  name                = "akscluster1"
  location            = "${azurerm_resource_group.aks.location}"
  resource_group_name = "${azurerm_resource_group.aks.name}"

  linux_profile {
    admin_username = "aksuser"

    ssh_key {
      key_data = "${file("../example_ssh_keypairs/ssh_public_cert.pem")}"
    }
  }

  agent_pool_profile {
    name            = "default"
    count           = 1
    vm_size         = "Standard_B2ms" # 2 Cores, 8 GB RAM $0.094/hour
    os_type         = "Linux"
    os_disk_size_gb = 30
  }

  service_principal {
    client_id     = "${var.service_principal_client_id}"
    client_secret = "${var.service_principal_client_secret}"
  }

  tags {
    Environment = "Dev"
  }
}
```

### Panic Output

```
Error: Error applying plan:

1 error(s) occurred:

* azurerm_kubernetes_cluster.aks: 1 error(s) occurred:

* azurerm_kubernetes_cluster.aks: containerservice.ManagedClustersClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="" Message=""
```

### Expected Behavior

The AKS resouces should have been created.

### Actual Behavior

Get an undescript error:

```
Failure sending request: StatusCode=400 -- Original Error: Code="" Message=""
```

### Steps to Reproduce

Leave off the `dns_prefix` field from the `azurerm_kubernetes_cluster` resource.

### References

According to the [Terraform documentation](https://www.terraform.io/docs/providers/azurerm/r/kubernetes_cluster.html), the `dns_prefix` field is optional. However, creating an AKS cluster with the Azure Portal, it is a required field.

**Comment by NickMRamirez** [comment]
It looks like the `dns_prefix` field is marked as optional in terraform-provider-azurerm/azurerm/resource_arm_kubernetes_cluster.go, line 38.

**Comment by katbyte** [comment]
Hey @NickMRamirez,

This was included in the 1.7 released available as of this afternoon, I hope it resolves this issue for you!
